tool/speed.cc

```diff
@@ -202,11 +196,20 @@ static bool SpeedRSA(const std::string &selected) {
     {"RSA 4096", kDERRSAPrivate4096, kDERRSAPrivate4096Len},
   };
 
+#if !defined(OPENSSL_BENCHMARK)
   for (unsigned i = 0; i < OPENSSL_ARRAY_SIZE(kRSAKeys); i++) {
+#else
+  for(unsigned i = 0; i < BM_ARRAY_SIZE(kRSAKeys); i++) {
+#endif
     const std::string name = kRSAKeys[i].name;
 
-    bssl::UniquePtr<RSA> key(
-        RSA_private_key_from_bytes(kRSAKeys[i].key, kRSAKeys[i].key_len));
+//    BM_NAMESPACE::UniquePtr<RSA> key(RSA_private_key_from_bytes(kRSAKeys[i].key, kRSAKeys[i].key_len));
+    /* d2i_RSAPrivateKey expects to be able to modify the input pointer as it parses the input data and we don't want it
+     * to modify the original *key data. Therefore create a new temp variable that points to the same data and pass
+     * in the reference to it. As a sanity check make sure input_key points to the end of the *key.
+     */
+    const uint8_t *input_key = kRSAKeys[i].key;
+    BM_NAMESPACE::UniquePtr<RSA> key(d2i_RSAPrivateKey(NULL, &input_key, (long) kRSAKeys[i].key_len));
```

Comment:
Since the signature is `TYPE *d2i_TYPE(TYPE **a, const unsigned char **ppin, long length);`, you are relying on that `uint8_t` is a typedef for `unsigned char`. Which it typically is.

Reply:
Unless I'm missing something, `d2i_RSAPrivateKey` takes a `uint8_t` though? Or am I looking at the wrong thing

Reply:
I see, AWS-LC and OpenSSL have different function signatures. Great.

I think this is fine because `uint8_t` is often the same as `unsigned char`.